cfme/utils/virtual_machines.py

```diff
@@ -73,9 +52,12 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
             vm_name = provider_crud.mgmt.deploy_template(timeout=timeout, **deploy_args)
             logger.info("Provisioned VM/instance %s", vm_name)  # instance ID in case of EC2
         except Exception as e:
-            logger.error('Could not provisioning VM/instance %s (%s: %s)',
+            logger.exception('Could not provisioning VM/instance %s (%s: %s)',
                 vm_name, type(e).__name__, str(e))
-            _vm_cleanup(provider_crud.mgmt, vm_name)
+            try:
+                provider_crud.mgmt.delete_vm(vm_name)
```

Comment:
I can't use VM.factory to call cleanup_on_provider() here because that would introduce a circular import (cfme.common.vm imports this file)

Reply:
i believe in that case a function local import may be allowed

Reply:
Agreed - but I think it can be added in a second PR, I already have a follow up cleanup